solutions/day_5/solution.py

```diff
@@ -0,0 +1,94 @@
+import copy
+from typing import List
+from functools import reduce
+
+
+TEXT_FILE_LOCATION = "input.txt"
+BATCH_SIZE = 3
+
+
+def edit_crate_holder(size: int) -> dict:
+    global CRATE_HOLDER
+    CRATE_HOLDER = {list_of_crates: [] for list_of_crates in range(1, size + 2)}
+
+
+def whitespace_helper(val: str) -> bool:
+    return val != " "
+
+
+def add_to_crate_holder(acc: None, val: List[str]) -> None:
+    if whitespace_helper(val[0][1]):
+        CRATE_HOLDER[1].append(val[0][1])
+    if whitespace_helper(val[0][5]):
+        CRATE_HOLDER[2].append(val[0][5])
+    if whitespace_helper(val[0][9]):
+        CRATE_HOLDER[3].append(val[0][9])
+    if whitespace_helper(val[0][13]):
+        CRATE_HOLDER[4].append(val[0][13])
+    if whitespace_helper(val[0][17]):
+        CRATE_HOLDER[5].append(val[0][17])
+    if whitespace_helper(val[0][21]):
+        CRATE_HOLDER[6].append(val[0][21])
+    if whitespace_helper(val[0][25]):
+        CRATE_HOLDER[7].append(val[0][25])
+    if whitespace_helper(val[0][29]):
+        CRATE_HOLDER[8].append(val[0][29])
+    if whitespace_helper(val[0][33]):
+        CRATE_HOLDER[9].append(val[0][33])
+
+
+def process(crates: dict, moves_to_be_made: List[str]) -> dict:
+    for val in moves_to_be_made:
+        i = int(val[1])
+        while i != 0:
+            part_one_crates[int(val[5])].append(part_one_crates[int(val[3])].pop())
+            i = i - 1
+    return part_one_crates
+
+
+def process_part_two(crates: dict, moves_to_be_made: List[str]) -> dict:
+    for val in moves_to_be_made:
+        i = int(val[1])
+        if i > 1:
```

Comment:
The logic for if `i == 1` is no different. You will pick up one crate and then reverse it which will still be itself. So this `if` is unnecessary. Also, use a `for _ in range(n)` to loop a set amount of times rather than this `i` decrementing business.